extensions/auth/opa/src/main/java/org/apache/polaris/extension/auth/opa/OpaAuthorizationConfig.java

```diff
@@ -0,0 +1,170 @@
+ */
+@ConfigMapping(prefix = "polaris.authorization.opa")
+public interface OpaAuthorizationConfig {
+  Optional<String> url();
+
+  Optional<String> policyPath();
+
+  Optional<AuthenticationConfig> auth();
+
+  Optional<HttpConfig> http();
+
+  /** Validates the complete OPA configuration */
+  default void validate() {
+    checkArgument(url().isPresent() && !url().get().isBlank(), "OPA URL cannot be null or empty");
+    checkArgument(
+        policyPath().isPresent() && !policyPath().get().isBlank(),
+        "OPA policy path cannot be null or empty");
+    checkArgument(auth().isPresent(), "Authentication configuration is required");
+
+    auth().get().validate();
+  }
+
+  /** HTTP client configuration for OPA communication. */
+  interface HttpConfig {
+    @WithDefault("2000")
+    int timeoutMs();
+
+    @WithDefault("true")
+    boolean verifySsl();
+
+    Optional<String> trustStorePath();
+
+    Optional<String> trustStorePassword();
+  }
+
+  /** Authentication configuration for OPA communication. */
+  interface AuthenticationConfig {
+    /** Type of authentication */
+    @WithDefault("none")
+    String type();
+
+    /** Bearer token authentication configuration */
+    Optional<BearerTokenConfig> bearer();
+
+    default void validate() {
+      switch (type()) {
+        case "bearer":
+          checkArgument(
+              bearer().isPresent(), "Bearer configuration is required when type is 'bearer'");
+          bearer().get().validate();
+          break;
+        case "none":
+          // No authentication - nothing to validate
+          break;
+        default:
+          throw new IllegalArgumentException(
+              "Invalid authentication type: " + type() + ". Supported types: 'bearer', 'none'");
+      }
+    }
+  }
+
+  interface BearerTokenConfig {
+    /** Type of bearer token configuration */
+    @WithDefault("static-token")
+    String type();
+
+    /** Static bearer token configuration */
+    Optional<StaticTokenConfig> staticToken();
+
+    /** File-based bearer token configuration */
+    Optional<FileBasedConfig> fileBased();
+
+    default void validate() {
+      switch (type()) {
+        case "static-token":
+          checkArgument(
+              staticToken().isPresent(),
+              "Static token configuration is required when type is 'static-token'");
+          staticToken().get().validate();
+          break;
+        case "file-based":
+          checkArgument(
+              fileBased().isPresent(),
+              "File-based configuration is required when type is 'file-based'");
+          fileBased().get().validate();
+          break;
+        default:
+          throw new IllegalArgumentException(
+              "Invalid bearer token type: " + type() + ". Must be 'static-token' or 'file-based'");
+      }
+    }
+
+    /** Configuration for static bearer tokens */
+    interface StaticTokenConfig {
+      /** Static bearer token value */
+      Optional<String> value();
```

Comment:
```suggestion
      String value();
```
This property has to be present when using the static token provider.

Reply:
Yeah I agree.

Let me take a look at this again, I'm new to the ConfigMapping library and I had trouble getting it to work without having the nested interfaces being type checked as well.